ods_ci/tests/Resources/Page/DistributedWorkloads/DistributedWorkloads.resource

```diff
@@ -256,3 +257,22 @@ Check missing Go test
     [Documentation]    Check that upstream Go test is not missing
     [Arguments]    ${test_run_output}
     Should Not Contain    ${test_run_output}    testing: warning: no tests to run    No Go tests were run
+
+Verify container images
+    [Documentation]    Verify container images
+    [Arguments]    ${pod_name}    ${container}    ${expected_image}
+    ${test_env}=  Is Test Enviroment ROSA-HCP
+    IF    ${test_env}==True
+        # We use Kyverno custom policies to pull unreleased images from quay registry for hypershift clusters
+        Log To Console    Verifying ${pod_name}'s container image is referred from quay.io
+        ${pod} =    Find First Pod By Name  namespace=${APPLICATIONS_NAMESPACE}   pod_regex=${pod_name}
+        Container Image Url Should Contain      ${APPLICATIONS_NAMESPACE}     ${pod}      ${container}
+        ...     quay.io/rhoai/${expected_image}
+        Log To Console    ${pod_name}'s container image is verified
+    ELSE
+        Log To Console    Verifying ${pod_name}'s container image is referred from registry.redhat.io
+        ${pod} =    Find First Pod By Name  namespace=${APPLICATIONS_NAMESPACE}   pod_regex=${pod_name}
+        Container Image Url Should Contain      ${APPLICATIONS_NAMESPACE}     ${pod}      ${container}
+        ...     registry.redhat.io/rhoai/${expected_image}
+        Log To Console    ${pod_name}'s container image is verified
+    END
```

Comment:
```suggestion
    ${test_env}=  Is Test Enviroment ROSA-HCP
        # We use Kyverno custom policies to pull unreleased images from quay registry for hypershift clusters

        ${registry_name}=   Set variable if ${test_env}==True
    ....    "quay.io"
    ....    "registry.redhat.io"
        Log To Console    Verifying ${pod_name}'s container image is referred from ${registry_name}
        ${pod} =    Find First Pod By Name  namespace=${APPLICATIONS_NAMESPACE}   pod_regex=${pod_name}
        Container Image Url Should Contain      ${APPLICATIONS_NAMESPACE}     ${pod}      ${container}
        ...     ${registry_name}/rhoai/${expected_image}
        Log To Console    ${pod_name}'s container image is verified
```

you can set the registry based on `test_env`

Reply:
Updated the code